storage/google_drive/drive_client_wrapper.py

```diff
@@ -90,9 +78,36 @@ def _get_folder_id(name, parent_id, recursive=False):
     log.info('Getting id of folder "{}" under parent with id "{}" - done. Folder id is "{}"'.format(name, parent_id, files[0].get('id')))
     return files[0].get('id')
 
-def _get_path_id(path, recursive=False):
+def _get_shared_folder_id(name):
+    log.info('Getting id of shared-with-me folder "{}"...'.format(name))
+    response = _drive_service.files().list(
+        q="name='{}' and sharedWithMe=true and mimeType='application/vnd.google-apps.folder'".format(name),
+        spaces='drive',
+        fields='files(id)').execute()
+    files = response.get('files', [])
+    if (len(files) == 0):
+        log.error('Folder "{}" not found in shared-with-me category.'.format(name))
+        exit(1)
+    if (len(files) > 1):
+        log.error('Multiple folders with name "{}" found in shared-with-me category.'.format(name))
+        exit(1)
```

Comment:
Could you put in an assert (len(files) == 1) which it will at this point
And then a:

folder = files[0]
folder_id = folder.get('id')

Which makes it clearer what this thing that we expected one of was

and use folder id below instead of using the repeated .get

Reply:
Thanks @lukechurch , I've done as you suggested here and in the function above as well as it was a similar situation.